packages/traceloop-sdk/traceloop/sdk/tracing/tracing.py

```diff
@@ -108,7 +109,64 @@ def __new__(
             if propagator:
                 set_global_textmap(propagator)
 
-            init_instrumentations()
+            instrument_set = False
+            if not instruments:
+                init_instrumentations()
+                instrument_set = True
+            else:
+                if 'openai' in instruments:
+                    init_openai_instrumentor()
+                    instrument_set = True
+                if 'anthropic' in instruments:
+                    init_anthropic_instrumentor()
```

Comment:
Potentially, the `init_` calls can return a boolean specifying whether they were able to instrument (for example, whether the `anthropic` library exists) and if not - log a warning.